runtime/context.go

```diff
@@ -281,54 +296,74 @@ type contextLogger struct {
 	skipZanzibarLogs bool
 }
 
-func (c *contextLogger) Debug(ctx context.Context, msg string, userFields ...zap.Field) {
+func (c *contextLogger) Debug(ctx context.Context, msg string, userFields ...zap.Field) context.Context {
 	c.log.Debug(msg, accumulateLogFields(ctx, userFields)...)
+	return ctx
 }
 
-func (c *contextLogger) Error(ctx context.Context, msg string, userFields ...zap.Field) {
+func (c *contextLogger) Error(ctx context.Context, msg string, userFields ...zap.Field) context.Context {
 	c.log.Error(msg, accumulateLogFields(ctx, userFields)...)
+	return ctx
 }
 
-func (c *contextLogger) Info(ctx context.Context, msg string, userFields ...zap.Field) {
+func (c *contextLogger) Info(ctx context.Context, msg string, userFields ...zap.Field) context.Context {
 	c.log.Info(msg, accumulateLogFields(ctx, userFields)...)
+	return ctx
 }
 
-func (c *contextLogger) Panic(ctx context.Context, msg string, userFields ...zap.Field) {
+func (c *contextLogger) Panic(ctx context.Context, msg string, userFields ...zap.Field) context.Context {
 	c.log.Panic(msg, accumulateLogFields(ctx, userFields)...)
+	return ctx
 }
 
-func (c *contextLogger) Warn(ctx context.Context, msg string, userFields ...zap.Field) {
+func (c *contextLogger) Warn(ctx context.Context, msg string, userFields ...zap.Field) context.Context {
 	c.log.Warn(msg, accumulateLogFields(ctx, userFields)...)
+	return ctx
 }
 
-func (c *contextLogger) DebugZ(ctx context.Context, msg string, userFields ...zap.Field) {
+func (c *contextLogger) DebugZ(ctx context.Context, msg string, userFields ...zap.Field) context.Context {
 	if !c.skipZanzibarLogs {
 		c.log.Debug(msg, accumulateLogFields(ctx, userFields)...)
+	} else {
+		ctx = accumulateLogMsgAndFieldsInContext(ctx, msg, userFields)
 	}
+	return ctx
 }
 
-func (c *contextLogger) ErrorZ(ctx context.Context, msg string, userFields ...zap.Field) {
+func (c *contextLogger) ErrorZ(ctx context.Context, msg string, userFields ...zap.Field) context.Context {
 	if !c.skipZanzibarLogs {
 		c.log.Error(msg, accumulateLogFields(ctx, userFields)...)
+	} else {
+		ctx = accumulateLogMsgAndFieldsInContext(ctx, msg, userFields)
 	}
+	return ctx
 }
 
-func (c *contextLogger) InfoZ(ctx context.Context, msg string, userFields ...zap.Field) {
+func (c *contextLogger) InfoZ(ctx context.Context, msg string, userFields ...zap.Field) context.Context {
 	if !c.skipZanzibarLogs {
 		c.log.Info(msg, accumulateLogFields(ctx, userFields)...)
+	} else {
+		ctx = accumulateLogMsgAndFieldsInContext(ctx, msg, userFields)
 	}
+	return ctx
 }
 
-func (c *contextLogger) PanicZ(ctx context.Context, msg string, userFields ...zap.Field) {
+func (c *contextLogger) PanicZ(ctx context.Context, msg string, userFields ...zap.Field) context.Context {
 	if !c.skipZanzibarLogs {
```

Comment:
Please invert the if condition to make it easier to read `!skip = do not skip = slightly hard to read`

```
if c.skipZanzibarLogs {
	ctx = accumulateLogMsgAndFieldsInContext(ctx, msg, userFields)	
} else {
	c.log.Panic(msg, accumulateLogFields(ctx, userFields)...)	
}
```

Reply:
Bump

Reply:
done